src/System.Management.Automation/engine/InternalCommands.cs

```diff
@@ -441,157 +717,172 @@ protected override void ProcessRecord()
+                        contextCmdlet: this,
+                        useLocalScope: false,
+                        errorHandlingBehavior: ScriptBlock.ErrorHandlingBehavior.WriteToCurrentErrorPipe,
+                        dollarUnder: InputObject,
+                        input: new object[] { InputObject },
+                        scriptThis: AutomationNull.Value,
+                        args: Array.Empty<object>());
+                }
             }
         }
 
+        private void InitScriptBlockParameterSet()
+        {
+            // Win8: 176403: ScriptCmdlets sets the global WhatIf and Confirm preferences
+            // This effects the new W8 foreach-object cmdlet with -whatif and -confirm
+            // implemented. -whatif and -confirm needed only for PropertyAndMethodSet
+            // parameter set. So erring out in cases where these are used with ScriptBlockSet.
+            // Not using MshCommandRuntime, as those variables will be affected by ScriptCmdlet
+            // infrastructure (wherein ScriptCmdlet modifies the global preferences).
+            Dictionary<string, object> psBoundParameters = this.MyInvocation.BoundParameters;
+            if (psBoundParameters != null)
+            {
+                SwitchParameter whatIf = false;
+                SwitchParameter confirm = false;
+
+                object argument;
+                if (psBoundParameters.TryGetValue("whatif", out argument))
+                {
+                    whatIf = (SwitchParameter)argument;
+                }
+
+                if (psBoundParameters.TryGetValue("confirm", out argument))
+                {
+                    confirm = (SwitchParameter)argument;
+                }
+
+                if (whatIf || confirm)
+                {
+                    string message = InternalCommandStrings.NoShouldProcessForScriptBlockSet;
+
+                    ErrorRecord errorRecord = new ErrorRecord(
+                        new InvalidOperationException(message),
+                        "NoShouldProcessForScriptBlockSet",
+                        ErrorCategory.InvalidOperation,
+                        null);
+                    ThrowTerminatingError(errorRecord);
+                }
+            }
+
+            // Calculate the start and end indexes for the processRecord script blocks
+            _end = _scripts.Count;
+            _start = _scripts.Count > 1 ? 1 : 0;
+
+            // and set the end script if it wasn't explicitly set with a named parameter.
+            if (!_setEndScript)
+            {
+                if (_scripts.Count > 2)
+                {
+                    _end = _scripts.Count - 1;
+                    _endScript = _scripts[_end];
+                }
+            }
+
+            // only process the start script if there is more than one script...
+            if (_end < 2)
+                return;
+
+            if (_scripts[0] == null)
+                return;
```

Comment:
```suggestion
           {    
                return;
            }
```

Reply:
This is existing code that I did not create, and I feel any style issues should be addressed in a separate Issue for this file.